File: src/talker.cpp
//导入ROS库文件
#include "ros/ros.h"
//导入ROS std_msgs标准消息包中的String消息
#include "std_msgs/String.h"
#include <sstream>

int main(int argc,char **argv)
{   
    //初始化ROS节点，节点名为talker
    ros::init(argc,argv,"talker");
    //声明ROS句柄
    ros::NodeHandle n;
    //创建一个发布器，向/chatter话题上发送std::msgs::String类型消息
    ros::Publisher chatter_pub = n.advertise<std_msgs::String>("chatter",1000);
    //创建一个ros::Rate对象，定时间隔为10hz
    ros::Rate loop_rate(10);

    int count = 0;
    while(ros::ok())
    {
        std_msgs::String msg;

        std::stringstream ss;
        ss << "hello world " << count;
        msg.data = ss.str();

        ROS_INFO("%s",msg.data.c_str());
        //使用发布器发布消息
        chatter_pub.publish(msg);
        //ros::spinOnce处理一次回调
        ros::spinOnce();
        //频率控制，控制循环频率为10hz
        loop_rate.sleep();
        ++count;
    }
    return 0;
}

File: scripts/talker.py
#!/usr/bin/env python 
#导入rospy库
import rospy
#导入ROS std_msgs标准消息中的String消息 
from std_msgs.msg import String 

def talker():
    #初始化ROS节点，命名节点为talker
    rospy.init_node('talker',anonymous=True)
    #创建发布器，向/chatter话题发布String类型消息
    pub = rospy.Publisher('chatter',String,queue_size=10)
    #创建一个rospy.Rate对象，控制循环频率为10hz
    rate = rospy.Rate(10)

    while not rospy.is_shutdown():
        #创建字符串
        hello_str = "hello world %s" % rospy.get_time()
        #打印字符串
        rospy.loginfo(hello_str)
        #利用发布器发布话题
        pub.publish(hello_str)
        #控制循环频率
        rate.sleep()

if __name__ == '__main__':
    try:
        talker()
    except rospy.ROSInterruptException:
        pass